cdc/sink/codec/canal.go

```diff
@@ -382,33 +462,14 @@ func (d *CanalEventBatchEncoder) Build() []*MQMessage {
 	if err != nil {
 		log.Fatal("Error when serializing Canal packet", zap.Error(err))
 	}
-	ret := NewMQMessage(nil, value, 0)
+	ret := NewMQMessage(nil, value, commitTs)
 	d.messages.Reset()
-	return []*MQMessage{ret}
-}
-
-// MixedBuild implements the EventBatchEncoder interface
-func (d *CanalEventBatchEncoder) MixedBuild(withVersion bool) []byte {
-	panic("Mixed Build only use for JsonEncoder")
-}
-
-// Size implements the EventBatchEncoder interface
-func (d *CanalEventBatchEncoder) Size() int {
-	// TODO: avoid marshaling the messages every time for calculating the size of the packet
-	err := d.refreshPacketBody()
-	if err != nil {
-		panic(err)
-	}
-	return proto.Size(d.packet)
-}
-
-// Reset implements the EventBatchEncoder interface
-func (d *CanalEventBatchEncoder) Reset() {
-	panic("Reset only used for JsonEncoder")
+	d.packet.Body = nil
```

Comment:
```suggestion
	d.packet.Body = d.packet.Body[:0]
```
Will this reduce some memory allocation

Reply:
Well!